.travis.yml

```diff
@@ -1,6 +1,51 @@
 language: csharp
 
-sudo: false  # use the new container-based Travis infrastructure 
+#dotnet cli require Ubuntu 14.04
+sudo: required
+dist: trusty
+
+#dotnet cli require OSX 10.10
+osx_image: xcode7.1
+
+addons:
+  apt:
+    packages:
+    - gettext
+    - libcurl4-openssl-dev
+    - libicu-dev
+    - libssl-dev
+    - libunwind8
+    - zlib1g
+
+os:
+  - osx
+
+env:
+  - CLI_VERSION="1.0.0.001494"
+
+before_install:
+  # Download script to install dotnet cli
+  - curl -L --create-dirs https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/install.sh -o ./scripts/obtain/install.sh
+  - find ./scripts -name "*.sh" -exec chmod +x {} \;
+  # use bash to workaround bug https://github.com/dotnet/cli/issues/1725
+  - sudo bash ./scripts/obtain/install.sh --channel "beta" --version "$CLI_VERSION"
+  - which dotnet
 
 script: 
   - ./build.sh All
+  # Build .NET Core with .NET CLI
+  # Workaround fsc doesnt work
+  - if test "$TRAVIS_OS_NAME" == "osx"; then sudo curl -L -s https://github.com/enricosada/fsharp-dotnet-cli-samples/raw/master/workarounds/1/osx.10.10-x64.zip | tar xvf - -C /usr/local/share/dotnet/cli/bin/ ; fi
+  # Workaround "Too many open files"
+  - ulimit -n 1024
+  # cli 
+  - dotnet restore
+  # Build Chessie
```

Comment:
If possibile, i dont want to change your build script.
If it's needed i'll do it obv, but as another pr, it's easier to review

There are two parts:

### install .NET CLI 

- the user may have it installed locally, no need to install always
- you can add an install script in your `scripts` directory, but it's annoying to maintain and it's two lines (the install script is downloaded from dotnet cli repo)
- if you want do that, use a `.dotnet` subdirectory instead of publish to default installation directory (i can help with that if you want, it's an argument of install script).
- usually the global .net cli install is enough, for shared ci is better a `.netcli` directory but appveyor/travis are always new images, so it's not important. But maybe it's better a `.netcli` subdir because it's easier to redo steps locally without change current global installation

### build and run with .NET CLI

- if possibile i dont want to touch your build script because i dont know enough of it. 
- i think the best is to add a fake task to run the same commands (not dotnet cli install, i mean `dotnet restore` and `dotnet build` ) /cc @forki for suggestions
- the current FAKE package creation task should include the binaries (you can use `dotnet build --output` to change directory for binaries if you want put it near current .NET 4.6 binaries)
- the new FAKE task should run conditionally i think because not everyone has .NET cli installed